internal/venice-common/src/main/java/com/linkedin/venice/kafka/validation/Segment.java

```diff
@@ -55,11 +55,11 @@ public class Segment {
   // Immutable state
   private final int partition;
   private final int segmentNumber;
-  private final CheckSum checkSum;
   private final Map<CharSequence, CharSequence> debugInfo;
   private final Map<CharSequence, Long> aggregates;
 
   // Mutable state
+  private final CheckSum checkSum;
```

Comment:
Why move this here? Because the content of the checksum is mutable, even though the pointer is immutable?